plugins/ssh-plugin/src/ssh-plugin-backend.ts

```diff
@@ -45,93 +55,109 @@ export async function start() {
     });
 }
 
-const getSshServiceName = async function (): Promise<string> {
-    /**
-     * Known Che services which can use the SSH key pairs.
-     */
-    const services: CheService[] = [
-        {
-            name: 'vcs',
-            displayName: 'VCS',
-            description: 'SSH keys used by Che VCS plugins'
-        },
-        {
-            name: 'machine',
-            displayName: 'Workspace Containers',
-            description: 'SSH keys injected into all Workspace Containers'
-        }
-    ];
+const getHostName = async function (): Promise<string> {
+    const hostName = await theia.window.showInputBox({ placeHolder: 'Please provide a Host name e.g. github.com' });
+    return hostName ? hostName : '';
+};
 
-    const option: theia.QuickPickOptions = {
-        matchOnDescription: true,
-        matchOnDetail: true,
-        canPickMany: false,
-        placeHolder: 'Select object:'
-    };
-    const sshServiceValue = await theia.window.showQuickPick<theia.QuickPickItem>(services.map(service =>
-        ({
-            label: service.displayName,
-            description: service.description,
-            detail: service.name,
-            name: service.name
-        })), option);
-    if (sshServiceValue) {
-        return Promise.resolve(sshServiceValue.label);
+const getKeyFilePath = function (name: string): string {
+    return resolve(os.homedir(), '.ssh', name.replace('.', '_'));
+};
+
+const updateConfig = async function (hostName: string): Promise<void> {
+    const sshDir = resolve(os.homedir(), '.ssh');
+    const configFile = resolve(sshDir, 'config');
+    await ensureFile(configFile);
+    await chmod(configFile, '644');
+    const keyConfig = `\nHost ${hostName.startsWith('default-') ? '*' : hostName}\nIdentityFile ${getKeyFilePath(hostName)}\n`;
+    const configContentBuffer = await readFile(configFile);
+    if (configContentBuffer.indexOf(keyConfig) >= 0) {
+        const newConfigContent = configContentBuffer.toString().replace(keyConfig, '');
```

Comment:
replace function is only replacing the first occurence, it's ok ?

Reply:
Yes, it is not expected that the config file has 2 identical key configs